### ISSUE TITLE
Add Dentistry schema to college of Dentistry website.

### ISSUE THREAD

**Issue by mjoneill** [uiowa/uiowa#4600]
As a search engine, I am able to consume schema.org/Dentist information on dentistry.uiowa.edu when I crawl the site.

### Resources/Background
* https://schema.org/Dentist
* https://www.drupal.org/project/schema_metatag/issues/3113067

### Problem

Currently, the dentistry site is not being understood as a dentist, and hence it does not appear in results for relevant queries.  In addition to evolving site content + working with Google My Business, etc...  there is an opportunity to help connect prospective patients with the dental services offered at the College of Dentistry by adding Dentist schema.

Note: I reference the schema_metatag project above. I am not necessarily requesting that approach. I would be fine hand coding this entire json-ld schema for this site, but there does not appear to be a way to add that.  With that said, it seems like we have flexibility/options on how we could implement Dentist schema.

**Comment by mjoneill** [comment]
Be aware, we will likely want to add dentist schema _AND_ CollegeorUniversity schema to this site, simultaneously. This could take the form of:
* both simultaneously to front page
* one on front page (collegeoruniveristy), one on an interior page (dentistry)
* something else

Right now it seems like first option is the recommended approach.

**Comment by briand44** [comment]
Check issue queues for ability to write own schema?

**Comment by briand44** [comment]
likely will require some investigation to figure out best way of implementing.

**Comment by briand44** [comment]
Michael willing to hand craft this if that is doable and no other more elegant options.

**Comment by briand44** [comment]
Michael to craft schema.org/Dentist JSON LD specific to College of Dentistry

**Comment by joewhitsitt** [comment]
https://stackoverflow.com/questions/30723531/best-json-ld-practices-using-multiple-script-elements

Didn't know you could have multiple on a page. Might be nice to have a "Additional schema" plain text bucket within SEO settings per page to then render out inside a separate script tag.

**Comment by richardbporter** [comment]
The module README seems to indicate a path for "adding groups and properties": https://git.drupalcode.org/project/schema_metatag/-/blob/8.x-1.x/README.txt

**Comment by mjoneill** [comment]
> Didn't know you could have multiple on a page.

Yep. There is reason for that confusion...at various times leaders in the space have variously advocated for and against that.  You can usually find a way to nest them, but some have gone mad trying to chase that ideal.  More often than not, having multiples is fine...until it isn't.  

For this one, there are a couple of approaches we could take in articulating CollegeorUniversity and Dentist on the same property, and we might want the flexibility a blank slate gives us to change if we get signals that one approach isn't working, etc...

**Comment by richardbporter** [comment]
> The module README seems to indicate a path for "adding groups and properties": https://git.drupalcode.org/project/schema_metatag/-/blob/8.x-1.x/README.txt

T-shaping on this issue a little bit.

**Comment by richardbporter** [comment]
I might be wrong here but this seems possible through the use of the schema_organization submodule and additional configuration. I'm able to produce this locally on the front page:
```
<script type="application/ld+json">{
    "@context": "https://schema.org",
    "@graph": [
        {
            "@type": "Dentist",
            "additionalType": "CollegeOrUniversity",
            "description": "This is the description.",
            "name": "This is the name.",
            "telephone": "555-555-5555",
            "contactPoint": {
                "@type": "PostalAddress",
                "telephone": "444-444-4444",
                "email": "foo@bar.com",
                "contactType": "Customer Service",
                "productSupported": "Dental Services",
                "areaServed": {
                    "@type": "State",
                    "name": "Iowa",
                    "address": {
                        "@type": "PostalAddress",
                        "streetAddress": "123 Foo St",
                        "addressLocality": "Iowa City",
                        "addressRegion": "IA",
                        "postalCode": "52242",
                        "addressCountry": "USA"
                    }
                }
            }
        }
    ]
}</script>
```

Is that the right direction? What other properties would need to be set? 

If this works, I can create a site split and enable + configure the required modules for dentistry. Unless schema_organization is something we want to enable for all sites.

**Comment by mjoneill** [comment]
@richardbporter: very nicely done, however, there are some issues we'd need to sort through.  Normally I'd recommend using `@graph` to have multiples of the same type.  I'm not sure it's necessary for this approach as written, but it doesn't appear to cause a problem. 

What is a bit of a problem is that this is saying that this dentist is also the dental college, and they have the same area served, same contact information, etc.  

I think we need a separate schema for CollegeorUniversity that defines the college, and we'd have a separate schema (even if it's on the same page) that defines Dentist.  Dentist would also have a relationship to CollegeorUniversity, likely as a "memberOf" attribute.  Let me get you an example of this.

**Comment by mjoneill** [comment]
OK.  Check it.  

So one page might have both these schemas on it, and they reference yet a third on a completely different website (uiowa.edu):  

```
<script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "CollegeOrUniversity",
  "@id": "https://tippie.uiowa.edu/#Tippie",
  "name": "Tippie College of Business",
  "alternateName": "University of Iowa Tippie College of Business",
  "url": "https://tippie.uiowa.edu",
  "logo": "https://tippie.uiowa.edu/sites/tippie.uiowa.edu/files/images/tippie-logo-square.png",
  "sameas":[
  "https://en.wikipedia.org/wiki/Tippie_College_of_Business",
  "https://www.wikidata.org/wiki/Q7808923",
  "https://www.linkedin.com/edu/university-of-iowa---henry-b.-tippie-college-of-business-18229",
  "https://www.facebook.com/TippieIowa",
  "https://twitter.com/tippieiowa",
  "https://www.instagram.com/tippiecollege/",
  "https://www.youtube.com/user/tippiebiz"
  ],
  "address": {
    "@type": "PostalAddress",
    "addressLocality": "Iowa City, Iowa",
    "postalCode": "52242",
    "streetAddress": "108 John Pappajohn Business Building"
  },
  "location": {
    "@type": "Place",
    "geo":{
      "@type": "GeoCoordinates",
      "latitude": "41.663109",
      "longitude": "-91.534926"
  }
},
  "areaServed": {
    "@type": "GeoCircle",
    "geoMidpoint": {
            "@type": "GeoCoordinates",
            "latitude": "41.663109",
            "longitude": "-91.534926"
          },
          "geoRadius": "804672"
  },
  "email": "tippie-business@uiowa.edu",
  "telephone": "800-553-4692",
  "foundingDate": "1921",
  "memberOf": {
  	"@type": "CollegeorUniversity",
    "@id": "https://uiowa.edu/#CollegeorUniversity"
  }
}
</script>
```

```
<script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "EducationalOrganization",
  "name": "Accounting Department",
  "alternateName": "Department of Accounting",
  "description": "The Accounting Department at Tippie College of Business produces distinguished scholars in accounting at the undergraduate, graduate, and PhD level. With a history of student success in academia and industry, Tippie is the place to be if you want to study finance at any level.",
  "address": {
    "@type": "PostalAddress",
    "name": [
      "Accounting Department",
      "W252 Pappajohn Business Building"
    ],
    "addressLocality": "Iowa City",
    "addressRegion": "IA",
    "postalCode": "52242",
    "streetAddress": "21 E Market St."
  },
  "contactPoint": [
    { "@type" : "ContactPoint",
      "telephone" : "+1-319-335-0910",
      "faxNumber" : "+1-319-335-1956",
      "contactType" : "customer support"
    }
  ],
  "makesOffer": [
    {
    "@type": "Offer",
    "name": "Master of Accountancy",
    "url": "https://tippie.uiowa.edu/future-graduate-students/masters-programs/accounting-mac"
},
{
"@type": "Offer",
"name": "PhD in Accounting",
"url": "https://tippie.uiowa.edu/future-graduate-students/phd-programs/phd-accounting"
},
{
"@type": "Offer",
"name": "BBA in Accounting",
"url": "https://tippie.uiowa.edu/future-undergraduates/majors/accounting-major"
}
],
  "potentialAction": {
    "@type": "ApplyAction",
    "target": "https://applytoiowa.force.com/ApplicantPortal/Portal_Login"
  },
    "parentOrganization": {
      "@id": "https://tippie.uiowa.edu/#Tippie"
    }
  }
  </script>
```


Now, drop that chunk in the schema validator and https://validator.schema.org and see what comes out the other end (validation, not rich snippets).  

If you look closely we could have even split this up further into separate schemas.  

Disclaimer, this is a very old implementation meant to provide an example of "multiple schemas coexisting on the same page or other pages, and establishing relationships between them with memberOf or parentOrganization, etc..."  Other attributes were likely experimental at the time and may or may not have aged well.

**Comment by mjoneill** [comment]
why are line breaks jacked? Hmmmm...

**Comment by mjoneill** [comment]
[schema_payload_example.json.txt](https://github.com/uiowa/uiowa/files/7854980/schema_payload_example.json.txt)

Adding a text file of the example above because readability.

**Comment by richardbporter** [comment]
I don't see the ability to add multiple schemas of the same type using the module. It also seems to use `@graph` by default and I don't know if that can be changed. It seems to be perfectly valid, as you mention. I pushed this out to dev if you want to log in and edit the front page there: https://dentistry.dev.drupal.uiowa.edu. You should see schema organization settings under the "SEO Settings" when editing a basic page.

If the module won't do exactly what we want and this schema won't change very often, we could probably inject the JSON+LD into the page(s) via code. I can investigate that.

**Comment by richardbporter** [comment]
>If the module won't do exactly what we want and this schema won't change very often, we could probably inject the JSON+LD into the page(s) via code. I can investigate that.

Definitely possible to hard-code it if that is easier.

**Comment by mjoneill** [comment]
> Definitely possible to hard-code it if that is easier.

I'd be interested in understanding this option a little better.  I think it might make sense for now.  Can it/could it work in tandem with the existing module?

**Comment by mjoneill** [comment]
> It also seems to use @graph by default and I don't know if that can be changed. It seems to be perfectly valid, as you mention.

Yep. It validates. I think it's valid, just a bit of an unexpected use.  I don't consider it a problem or something we should invest in changing.

> I pushed this out to dev if you want to log in and edit the front page there

I will take a look and let you know what I see.  Thanks!

**Comment by mjoneill** [comment]
OK.  I have two schemas.  They both _could_ live on the front page of dentistry.uiowa.edu, but I'd like to avoid loading all the clinical schema on every pageload for front page of the dentistry site. 

Is it possible to load the CollegeorUniversity schema at https://dentistry.uiowa.edu and the Dentist schema at https://dentistry.uiowa.edu/patient-care/? 

- CollegeorUniversity: https://iowa-my.sharepoint.com/:u:/g/personal/mjoneill_uiowa_edu/EfEbwqIw1dVBrALRn9jB2pIBVcBqeio6vjRFUnQvU515Zg?e=ABrok4
- Dentist: https://iowa-my.sharepoint.com/:u:/g/personal/mjoneill_uiowa_edu/EcgqolDKISpBgTJEAg9yUFABVuXjrkmNKuy7FciQ5o3Y7w?e=PwidNl

Note: Both should validate right now, but are not "done." I have some additional tweaking depending on what page they can live on + some answers I'm waiting on from Viki on, etc...  They will have further revisions.

**Comment by joewhitsitt** [comment]
Talked to Rick about this, @mjoneill can we push back on the specific path for the Dentist schema and place it on all pages but the front?

**Comment by mjoneill** [comment]
> all pages but the front

No. I'll do some reading on it, but that's an important piece to this.  UniversityorCollege schema should be on just the front page. In the case of schema.org/Dentist, just the front page of the clinical section (if possible).  

When it's on the front page, it's saying, "This website is about this thing defined in the schema."  When it's on an interior page, it's saying, "This page is about this thing defined in the schema."  

Many popular schema plugins in the Wordpress realm get this wrong, and add the same schema to every page.  That's badnewsbears.

**Comment by joewhitsitt** [comment]
>No. I'll do some reading on it, but that's an important piece to this.

I will implement as you suggest.

**Comment by joewhitsitt** [comment]
@mjoneill the current hardcoded setup is now on dev for review:

https://validator.schema.org/#url=https%3A%2F%2Fdentistry.dev.drupal.uiowa.edu
https://validator.schema.org/#url=https%3A%2F%2Fdentistry.dev.drupal.uiowa.edu%2Fpatient-care

**Comment by mjoneill** [comment]
Any chance of changing the URL line of the patient-care schema to read: 
-   "url": "https://dentistry.uiowa.edu/patient-care",
instead of 
-   "url": "https://dentistry.uiowa.edu",


If not, we'll pick it up on the next push when we get info from Viki.

^^^ I made the change in the source files on onedrive so we won't lose it in the future.

**Comment by mjoneill** [comment]
I saw this round tripping to blocked/waiting and back.  Let me know if we need to discuss something.

**Comment by joewhitsitt** [comment]
@mjoneill, your last comment made it sound like you were good with it as a first step. i made the URL change (not deployed to DEV) and have asked the developers for a technical review/approval.

**Comment by mjoneill** [comment]
> @mjoneill, your last comment made it sound like you were good with it as a first step. 

Yes. I absolutely am. I just wanted to make sure I wasn't blocking something (missed standup this AM).
